packages/core/src/app/shipping/SingleShippingForm.tsx

```diff
@@ -111,10 +112,50 @@ class SingleShippingForm extends PureComponent<
     ) {
         super(props);
 
-        const { updateAddress } = this.props;
-
         this.debouncedUpdateAddress = debounce(
             async (address: Address, includeShippingOptions: boolean) => {
+                const { 
+                    updateAddress,
+                    shippingAddress,
+                    values: { shippingAddress: addressForm },
+                    moveCheckUpdatedAddressValueWithinDebouncedExperiment
+                } = this.props;
+
+                if (moveCheckUpdatedAddressValueWithinDebouncedExperiment) {
+                    const updatedShippingAddress = addressForm && mapAddressFromFormValues(addressForm);
+            
+                    if (Array.isArray(shippingAddress?.customFields)) {
+                        includeShippingOptions = !isEqual(
+                            shippingAddress?.customFields,
+                            updatedShippingAddress?.customFields
+                        ) || includeShippingOptions;
+                    }
+            
+                    if (!updatedShippingAddress || isEqualAddress(updatedShippingAddress, shippingAddress)) {
+                        return;
+                    }
+                    
+                    this.setState({ isUpdatingShippingData: true });
+
+                    try {
+                        await updateAddress(updatedShippingAddress, {
+                            params: {
+                                include: {
+                                    'consignments.availableShippingOptions': includeShippingOptions,
+                                },
+                            },
+                        });
+    
+                        if (includeShippingOptions) {
+                            this.setState({ hasRequestedShippingOptions: true });
+                        }
+                    } finally {
+                        this.setState({ isUpdatingShippingData: false });
```

Comment:
Do we need this https://github.com/bigcommerce/checkout-js/pull/2023/files#diff-738bdd4ad4c50c9374ff90b1691ec572ef3d86fe8457f2d942b1c7af745df69aR175?